src/RestServer/Controllers/v1/LedgerController.cs

```diff
@@ -0,0 +1,389 @@
+        {
+            var block = NativeContract.Ledger.GetBlock(_neosystem.StoreView, blockIndex);
+            if (block == null)
+                throw new BlockNotFoundException(blockIndex);
+            return Ok(block);
+        }
+
+        /// <summary>
+        /// Gets a block header by block index.
+        /// </summary>
+        /// <param name="blockIndex" example="0">Blocks index.</param>
+        /// <returns>Block Header Object.</returns>
+        /// <response code="200">Successful</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpGet("blocks/{index:min(0)}/header", Name = "GetBlockHeader")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(Header))]
+        public IActionResult GetBlockHeader(
+            [FromRoute(Name = "index")]
+            uint blockIndex)
+        {
+            var block = NativeContract.Ledger.GetBlock(_neosystem.StoreView, blockIndex);
+            if (block == null)
+                throw new BlockNotFoundException(blockIndex);
+            return Ok(block.Header);
+        }
+
+        /// <summary>
+        /// Gets the witness of the block
+        /// </summary>
+        /// <param name="blockIndex" example="0">Block Index.</param>
+        /// <returns>Witness Object</returns>
+        /// <response code="200">Successful</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpGet("blocks/{index:min(0)}/witness", Name = "GetBlockWitness")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(Witness))]
+        public IActionResult GetBlockWitness(
+            [FromRoute(Name = "index")]
+            uint blockIndex)
+        {
+            var block = NativeContract.Ledger.GetBlock(_neosystem.StoreView, blockIndex);
+            if (block == null)
+                throw new BlockNotFoundException(blockIndex);
+            return Ok(block.Witness);
+        }
+
+        /// <summary>
+        /// Gets the transactions of the block.
+        /// </summary>
+        /// <param name="blockIndex" example="0">Block Index.</param>
+        /// <param name="skip">Page</param>
+        /// <param name="take">Page Size</param>
+        /// <returns>An array of transaction object.</returns>
+        /// <response code="200">Successful</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpGet("blocks/{index:min(0)}/transactions", Name = "GetBlockTransactions")]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(Transaction[]))]
+        public IActionResult GetBlockTransactions(
+            [FromRoute(Name = "index")]
+            uint blockIndex,
+            [FromQuery(Name = "page")]
+            int skip = 1,
+            [FromQuery(Name = "size")]
+            int take = 1)
```

Comment:
```suggestion
            int take = 50)
```
I'm not sure, but how about make the default count the same as default MaxPageSize or maybe a value between 1 and 50. Even I have only 2 transactions, I still need input page, size into request, seems not convenient.

Reply:
I did that so if they make the request without `page` or `size` it only gets one. So web crawlers don't eat bandwidth. But sure we can change them all.